src/react/hoc/subscription-hoc.tsx

```diff
@@ -17,8 +17,8 @@ import { OperationOption, OptionProps, DataProps } from './types';
 export function withSubscription<
   TProps extends TGraphQLVariables | {} = {},
   TData = {},
-  TGraphQLVariables = {},
-  TChildProps = DataProps<TData, TGraphQLVariables>
+  TGraphQLVariables extends {} = {},
+  TChildProps extends { [x: string]: any; } = DataProps<TData, TGraphQLVariables>
```

Comment:
[nit] You can use `Record` here if you prefer the stylistic cleanliness instead of the indexed type:

```suggestion
  TChildProps extends Record<string, any> = DataProps<TData, TGraphQLVariables>
```